example/Android/android-build.sh

```diff
@@ -0,0 +1,106 @@
+#!/bin/bash
+set -ex
+if [ -d lib ]; then
+	echo "Please run this script from an empty directory"
+	exit 1
+fi
+if [[ ( "$#" -ne 4 ) && ( "$#" -ne 5 ) ]]; then
+    echo "Illegal number of parameters"
+    echo "Usage: $0 <PLATFORM> <ANDROID_VERSION> <NDK_PATH> <PACKAGE> [BUILD_TYPE=Release]"
+    echo "Example: $0 arm64-v8a 26 /Users/me/Downloads/android-ndk-r16b jp.co.soramitsu.iroha.android Debug"
+    exit 1
+fi
+
+PLATFORM=$1
+VERSION=$2
+NDK_PATH=$3
+PACKAGE=$4
+BUILD_TYPE=$5
+
+LIBP=lib
+case "$PLATFORM" in
+  armeabi)
+    ARCH=arch-arm
+    ;;
+  armeabi-v7a)
+    ARCH=arch-arm
+    ;;
+  arm64-v8a)
+    ARCH=arch-arm64
+    ;;
+  x86)
+    ARCH=arch-x86
+    ;;
+  x86_64)
+    ARCH=arch-x86_64
+    LIBP=lib64
+    ;;
+  *)
+	echo Wrong ABI name: "$PLATFORM"
+    exit 1
+    ;;
+esac
+
+if [[ ( -z ${BUILD_TYPE} ) || (${BUILD_TYPE} = Release) ]]; then
+    PROTOBUF_LIB_NAME=protobuf
+elif [[ ${BUILD_TYPE} = Debug ]]; then
+    PROTOBUF_LIB_NAME=protobufd
+else
+    echo "Unknown build type: ${BUILD_TYPE}"
+    exit 1
+fi
+
+ANDROID_TOOLCHAIN_ARGS=(-DCMAKE_SYSTEM_NAME=Android -DCMAKE_SYSTEM_VERSION="$VERSION" -DCMAKE_ANDROID_ARCH_ABI="$PLATFORM" -DANDROID_NDK="$NDK_PATH" -DCMAKE_ANDROID_STL_TYPE=c++_static)
+DEPS_DIR="$PWD"/iroha/dependencies
+INSTALL_ARGS=(-DCMAKE_INSTALL_PREFIX="$DEPS_DIR")
+
+CORES=$(getconf _NPROCESSORS_ONLN)
+if [ "$CORES" -gt 1 ]; then
+    CORES=$((CORES-1))
+fi
+
+# iroha develop
+git clone -b develop --depth=1 https://github.com/hyperledger/iroha
+mkdir "$DEPS_DIR"
+mkdir "$DEPS_DIR"/lib
+mkdir "$DEPS_DIR"/include
+
+# boost
+wget https://dl.bintray.com/boostorg/release/1.66.0/source/boost_1_66_0.tar.gz
+tar xf ./boost_1_66_0.tar.gz
+cp -R ./boost_1_66_0/boost "$DEPS_DIR"/include
+
+# protobuf
+git clone https://github.com/google/protobuf
+(cd ./protobuf ; git checkout b5fbb742af122b565925987e65c08957739976a7)
+cmake -Dprotobuf_BUILD_TESTS=OFF -DCMAKE_BUILD_TYPE="$BUILD_TYPE" -H./protobuf/cmake -B./protobuf/host_build # build for host to get js_embed
+VERBOSE=1 cmake --build ./protobuf/host_build -- -j"$CORES"
+sed -i.bak "s~COMMAND js_embed~COMMAND \"$PWD/protobuf/host_build/js_embed\"~" ./protobuf/cmake/libprotoc.cmake
+
+LDFLAGS="-llog -landroid" cmake "${ANDROID_TOOLCHAIN_ARGS[@]}" "${INSTALL_ARGS[@]}" -Dprotobuf_BUILD_TESTS=OFF -DCMAKE_BUILD_TYPE="$BUILD_TYPE" -H./protobuf/cmake -B./protobuf/.build
+VERBOSE=1 cmake --build ./protobuf/.build --target install -- -j"$CORES"
+
+# ed25519
+git clone git://github.com/hyperledger/iroha-ed25519
+(cd ./iroha-ed25519 ; git checkout e7188b8393dbe5ac54378610d53630bd4a180038)
+cmake "${ANDROID_TOOLCHAIN_ARGS[@]}" "${INSTALL_ARGS[@]}" -DTESTING=OFF -DCMAKE_BUILD_TYPE="$BUILD_TYPE" -DBUILD=STATIC -H./iroha-ed25519 -B./iroha-ed25519/build
+VERBOSE=1 cmake --build ./iroha-ed25519/build --target install -- -j"$CORES"
+mv "$DEPS_DIR"/lib/static/libed25519.a "$DEPS_DIR"/lib; rmdir "$DEPS_DIR"/lib/static/
```

Comment:
Too hackish, maybe just add patch to Finded25519?

Reply:
I think patching Finded25519 is even more hackish :) Actually this way should be more reliable and resistant to changes in Finded25519.